packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PreferencesStep/PreferencesStep.tsx

```diff
@@ -1,24 +1,76 @@
+import app from 'client/scripts/state';
+import {
+  useFetchSelfProfileQuery,
+  useUpdateProfileByAddressMutation,
+} from 'client/scripts/state/api/profiles';
 import {
   PreferenceTags,
   usePreferenceTags,
 } from 'client/scripts/views/components/PreferenceTags';
 import { CWText } from 'client/scripts/views/components/component_kit/cw_text';
 import { CWButton } from 'client/scripts/views/components/component_kit/new_designs/CWButton';
-import React from 'react';
+import AddressInfo from 'models/AddressInfo';
+import React, { useEffect, useRef } from 'react';
 import './PreferencesStep.scss';
 
 type PreferencesStepProps = {
   onComplete: () => void;
 };
 
 const PreferencesStep = ({ onComplete }: PreferencesStepProps) => {
-  const { selectedTags, toggleTagFromSelection } = usePreferenceTags({});
+  const initialTagsSet = useRef(false);
+  const { preferenceTags, setPreferenceTags, toggleTagFromSelection } =
+    usePreferenceTags({});
+
+  const { mutateAsync: updateProfile, isLoading: isUpdatingProfile } =
+    useUpdateProfileByAddressMutation();
+
+  const { data: profile, isLoading } = useFetchSelfProfileQuery({
+    apiCallEnabled: true,
+  });
+
+  useEffect(() => {
+    if (!isLoading && profile && !initialTagsSet.current) {
+      const profileTags = profile.tags;
+      setPreferenceTags((tags) =>
+        [...(tags || [])].map((t) => ({
+          ...t,
+          isSelected: !!profileTags.find((pt) => pt.id === t.item.id),
+        })),
+      );
+      profile.addresses &&
+        app.user.setAddresses(
+          profile.addresses.map(
+            (a) =>
+              new AddressInfo({
+                address: a.address,
+                communityId: a.community_id,
+                id: a.id,
+                ghostAddress: a.ghost_address,
+                keytype: a.keytype,
+                lastActive: a.last_active,
+                profileId: a.profile_id,
+                walletId: a.wallet_id,
+                walletSsoSource: a.wallet_sso_source,
+              }),
+          ),
+        );
```

Comment:
I am not fan of doing this kind of modifications here in the component. Can't we do this in query on success?

Reply:
we can, will move it there

Reply:
[done](https://github.com/hicommonwealth/commonwealth/pull/7824/commits/cfdb72351985ecb84de6e6b9a1357c9d2cc6ebea)